src/main/java/me/redstom/privaterooms/entities/entity/model/ModelSynchronizerService.java

```diff
@@ -0,0 +1,151 @@
+        VoiceChannel channel = voiceChannelManager.getChannel();
+        update = update.andThen(model -> model.visibility(visibility));
+
+        voiceChannelManager.putPermissionOverride(
+          channel.getGuild().getPublicRole(),
+          visibility.allowRaw(),
+          visibility.denyRaw()
+        );
+
+        actions.add("Set channel visibility to %s".formatted(visibility.name()));
+        channelName(room.model().channelName());
+        return this;
+    }
+
+    public ModelSynchronizerService addUserTo(User user, ModelEntityType modelEntityType) {
+        return switch (modelEntityType) {
+            case WHITELIST -> addUserToWhitelist(user);
+            case BLACKLIST -> addUserToBlacklist(user);
+            case MODERATOR -> addUserToModerator(user);
+        };
+    }
+
+    public ModelSynchronizerService addUserToWhitelist(User user) {
+        Member member = voiceChannelManager.getGuild().retrieveMemberById(user.discordId()).complete();
+
+        addUserToModel(user, ModelEntityType.WHITELIST);
+        putPermissionTo(member, ModelEntityType.WHITELIST);
+        actions.add("Added %s (%s) to the whitelist".formatted(member.getEffectiveName(), member.getId()));
+        return this;
+    }
+
+    public ModelSynchronizerService addUserToBlacklist(User user) {
+        Member member = voiceChannelManager.getGuild().retrieveMemberById(user.discordId()).complete();
+
+        addUserToModel(user, ModelEntityType.BLACKLIST);
+        putPermissionTo(member, ModelEntityType.BLACKLIST);
+        actions.add("Added %s (%s) to the blacklist".formatted(member.getEffectiveName(), member.getId()));
+        return this;
+    }
+
+    public ModelSynchronizerService addUserToModerator(User user) {
+        Member member = voiceChannelManager.getGuild().retrieveMemberById(user.discordId()).complete();
+
+        addUserToModel(user, ModelEntityType.MODERATOR);
+        putPermissionTo(member, ModelEntityType.MODERATOR);
+        actions.add("Added %s (%s) as a moderator".formatted(member.getEffectiveName(), member.getId()));
+        return this;
+    }
+
+    private void addUserToModel(User user, ModelEntityType modelEntityType) {
+        update = update.andThen(model -> addUserToModel(model, ModelUser.builder()
+          .referringUser(user)
+          .type(modelEntityType)
+          .build())
+        );
+    }
+
+    private Model addUserToModel(Model model, ModelUser modelUser) {
+        model.users().add(modelUser);
+        return model;
+    }
+
+    private void putPermissionTo(IPermissionHolder permissionHolder, PermissionSet permissionSet) {
+        voiceChannelManager.putPermissionOverride(
+          permissionHolder,
+          permissionSet.allowRaw(),
+          permissionSet.denyRaw()
+        );
+    }
+
+    public ModelSynchronizerService reason(String reason) {
+        voiceChannelManager.reason(reason);
+        return this;
+    }
```

Comment:
Add a default reason maybe by a factory constructor asking the issuer of the modification.